Sources/SwiftGRPC/Core/Channel.swift

```diff
@@ -21,19 +21,31 @@ import Foundation
 /// A gRPC Channel
 public class Channel {
   private let mutex = Mutex()
+  /// Weak references to API calls using this channel that are in-flight
+  private let activeCalls = NSHashTable<Call>.weakObjects()
```

Comment:
Definitely open to suggestions here. Some other options I had considered:
1. Instead storing a `private var shutdownCallbacks: [() -> Void]()` and adding a `register` function to `Channel` which would then be called where the `Call` is instantiated outside of the `Channel`. This felt a bit wasteful since that means each call would have to be manually registered with a channel, and even after a call completes, a long-lived channel would end up with a large list of closures
2. To solve the large list of closures in `#1`, could also add an `unregister` function for a given call which is called when the call is deallocated, but that would get a bit messy with all the required state tracking

The solution I stuck with (`NSHashTable` with `weak` references) seemed like the cleanest

Reply:
Unfortunately, `NSHashTable` is not available on Linux. Another alternative could be to have something like this:

```
// Might need to use `class` instead of `struct`.
struct WeakReference<T: class> {
  weak var value: T?

  init(value: T) {
    self.value = value
  }
}

...

private var activeCalls: [WeakReference<Call>] = []
```

and prune out `nil` references (e.g. via `self.activeCalls = self.activeCalls.filter { $0.value != nil }`) on a regular basis, e.g. every 60 seconds.

Or, do we actually even need this at all? Maybe gRPC-Core automatically shuts down all calls when the corresponding channel has been destroyed, so we wouldn't need to do anything?

Reply:
> Another alternative could be to have something like this:

Yes, that could work. I can update to use that functionality. 

> prune out `nil` references (e.g. via `self.activeCalls = self.activeCalls.filter { $0.value != nil }`) on a regular basis, e.g. every 60 seconds

Could do that. Not a huge fan of having to run a "cleanup" task, but the only other viable alternatives that come to mind are doing it every time a new call is made (which is probably too often) or every X times a new call is made 😕 

> do we actually even need this at all? Maybe gRPC-Core automatically shuts down all calls when the corresponding channel has been destroyed, so we wouldn't need to do anything?

Unfortunately, although gRPC-Core might be shutting down the underlying calls, it doesn't see, to notify us about them as far as I can tell. For example, if I remove the `self.activeCalls.allObjects.forEach { $0.cancel() }` line, the `testCancelsActiveCallWhenShutdownIsCalled` test fails. I think we're stuck with doing it manually unless I'm missing something

Reply:
Updated if you can take another look 🙏